apps/frontend/src/app/Data/Weapons/Polearm/ProspectorsDrill/index.tsx

```diff
@@ -0,0 +1,71 @@
+import { allElementKeys, type WeaponKey } from '@genshin-optimizer/consts'
+import { allStats } from '@genshin-optimizer/gi-stats'
+import { objKeyMap, objKeyValMap, range } from '@genshin-optimizer/util'
+import { input } from '../../../../Formula'
+import { lookup, naught, prod, subscript } from '../../../../Formula/utils'
+import { cond, st, stg, trans } from '../../../SheetUtil'
+import type { IWeaponSheet } from '../../IWeaponSheet'
+import WeaponSheet, { headerTemplate } from '../../WeaponSheet'
+import { dataObjForWeaponSheet } from '../../util'
+
+const key: WeaponKey = 'ProspectorsDrill'
+const data_gen = allStats.weapon.data[key]
+const [, trm] = trans('weapon', key)
+
+const atk_Inc = [-1, 0.03, 0.04, 0.05, 0.06, 0.07]
+const allEleInc = [-1, 0.07, 0.085, 0.1, 0.115, 0.13]
+const [condStackPath, condStack] = cond(key, 'Struggle')
+const atk_ = lookup(
+  condStack,
+  objKeyMap(range(1, 3), (i) =>
```

Comment:
The range should be stored and reused in a const like marksConsumedArr or similar